src/core/ErrorLoggingDIspatcher.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+
+namespace CR.MessageDispatch.Core
+{
+    public class ErrorLoggingDispatcher<TMessage> : IDispatcher<TMessage>
+    {
+        private readonly IDispatcher<TMessage> _dispatcher;
+        private readonly Action<TMessage, Exception> _exceptionLogAction;
```

Comment:
That was for the retry, for logging an exception it is possible the logging code will want to include properties from the event type.